appcelerator/titanium_mobile
# [TIMOB-24642] - Correct require implementation

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24642

*Note* iOS does not like a leading slash on paths, so '/node_modules' in the below is 'node_modules' on iOS

- Decrement i before continuing
- If startDir/path === '/' return early with "/node_modules" to prevent unnecessary loop
- Always appends '/node_modules'
- On iOS correct the path builder to split the array rather than perform a substring on the string

Test case:


1. Download the test app linked in the ticket description
2. Build the app, it should load to a yellow screen and in the console you should see.

```
[INFO]  boz index.js loaded
[INFO]  boo boo.js loaded
[INFO]  baz index.js loaded
[INFO]  bar bar.js loaded
[INFO]  foo lib/foo.js loaded
[INFO]  foo index.js loaded
```
